src/useStorage/storage.ts

```diff
@@ -0,0 +1,92 @@
+import { globalExists } from '../utilities/global'
+
+export type StorageType = 'session' | 'local'
+
+export class StorageManager {
+  private readonly type: StorageType
+
+  public get length(): number {
+    const storage = this.storage()
+
+    if (!storage) {
+      return 0
+    }
+
+    return storage.length
+  }
+
+  public constructor(type: StorageType) {
+    this.type = type
+  }
+
+  public get<T>(key: string): T | null
+  public get<T>(key: string, defaultValue: T): T
+  public get<T>(key: string, defaultValue: T | null = null): T | null {
+    const storage = this.storage()
+
+    if (!storage) {
+      return null
+    }
+
+    const value = storage.getItem(key)
+
+    if (value === null) {
+      return defaultValue
+    }
+
+    return JSON.parse(value) as T
+  }
+
+  public set<T>(key: string, value: T): void {
+    const string = JSON.stringify(value)
+    const storage = this.storage()
+
+    if (!storage) {
+      return
+    }
+
+    return storage.setItem(key, string)
+  }
+
+  public remove(key: string): void {
+    const storage = this.storage()
+
+    if (!storage) {
+      return
+    }
+
+    return storage.removeItem(key)
+  }
+
+  public clear(): void {
+    const storage = this.storage()
+
+    if (!storage) {
+      return
+    }
+
+    return storage.clear()
+  }
+
+  public key(index: number): string | null {
+    const storage = this.storage()
+
+    if (!storage) {
+      return null
+    }
+
+    return storage.key(index)
+  }
+
+  private storage(): Storage | null {
+    if (this.type === 'local' && globalExists('localStorage')) {
+      return localStorage
+    }
+
+    if (this.type === 'session' && globalExists('sessionStorage')) {
+      return sessionStorage
+    }
+
+    return null
+  }
+}
```

Comment:
I'm surprised to see this is not a computed getter?

Reply:
I suppose it probably could be. But my intent was to not compute this until it's needed.